src/riak_object.erl

```diff
@@ -1064,11 +1073,34 @@ fold_meta_to_bin(Key, Value, {{_Vt,_Del,_Lm}=Elems,RestBin}) ->
     MetaBin = <<KeyLen:32/integer, KeyBin/binary, ValueLen:32/integer, ValueBin/binary>>,
     {Elems, <<RestBin/binary, MetaBin/binary>>}.
 
+%% @doc Encode the contents of a riak object, either using
+%% term_to_binary (if Enc == erlang), or msgpack encoding (if Enc ==
+%% msgpack)
+encode(Bin, Enc) when Enc == erlang ->
+    encode_maybe_binary(Bin);
+encode(Bin, Enc) when Enc == msgpack ->
+    encode_msgpack(Bin).
+
+encode_msgpack(Bin) ->
+    msgpack:pack(Bin, [{format, jsx}]).
```

Comment:
It's not clear for me - `msgpack:pack/2` only accepts `msgpack:object()` type and returns error for other invalid inputs. How do we make sure this function only have packable terms?